src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -337,9 +338,49 @@ private IEnumerable<CompletionItem> GetOutputValueCompletions(SemanticModel mode
             return GetValueCompletionsForType(declaredType, context.ReplacementRange, model, context, loopsAllowed: true);
         }
 
-        private IEnumerable<CompletionItem> GetResourceOrModuleBodyCompletions(BicepCompletionContext context)
+        private IEnumerable<CompletionItem> GetResourceBodyCompletions(BicepCompletionContext context)
         {
-            if (context.Kind.HasFlag(BicepCompletionContextKind.ResourceBody) || context.Kind.HasFlag(BicepCompletionContextKind.ModuleBody))
+            if (context.Kind.HasFlag(BicepCompletionContextKind.ResourceBody))
+            {
+                yield return CreateResourceBodyCompletion(context);
+
+                yield return CreateResourceOrModuleConditionCompletion(context.ReplacementRange);
+
+                // loops are always allowed in a resource/module
+                foreach (var completion in CreateLoopCompletions(context.ReplacementRange, LanguageConstants.Object, filtersAllowed: true))
+                {
+                    yield return completion;
+                }
+            }
+        }
+
+        private CompletionItem CreateResourceBodyCompletion(BicepCompletionContext context)
+        {
+            StringSyntax? stringSyntax = (context.EnclosingDeclaration as ResourceDeclarationSyntax)?.TypeString;
+
+            if (stringSyntax is not null)
+            {
+                string type = stringSyntax.StringTokens[0].Text;
```

Comment:
>string type = stringSyntax.StringTokens[0].Text; [](start = 16, length = 48)

This logic will break when with nested resource types unfortunately because those allow "relative" type names. My recommendation would be to do call `GetDeclaredType() on the resource declaration. If that produces a `ResourceTypeReference`, you can get the parsed type from it. If not, the type string in the declaration must be invalid.

Reply:
Updated here:
https://github.com/Azure/bicep/commit/e97046a2c43eedb396d40cdb0ad18b0f8ed6c250

Reply:
Marking this resolved. Let me know if you have any questions.